pkg/cmd/server/api/types.go

```diff
@@ -132,15 +132,21 @@ type NodeConfig struct {
 	IPTablesSyncPeriod string
 
 	// VolumeConfig contains options for configuring volumes on the node.
-	VolumeConfig VolumeConfig
+	VolumeConfig NodeVolumeConfig
 }
 
-// VolumeConfig contains options for configuring volumes on the node.
-type VolumeConfig struct {
+// NodeVolumeConfig contains options for configuring volumes on the node.
+type NodeVolumeConfig struct {
 	// LocalQuota contains options for controlling local volume quota on the node.
 	LocalQuota LocalQuota
 }
 
+// MasterVolumeConfig contains options for configuring volume plugins in the master node.
+type MasterVolumeConfig struct {
+	// EnableDynamicProvisioning is a boolean that toggles the dynamic provisioning feature on/off
+	EnableDynamicProvisioning bool
```

Comment:
Opt-ing in, I think, generally makes more sense.  Perhaps this was done wrongly upstream in that this is automatically on by default if you have a cloud provider configured.  Great UX: "No configuration required!", but now we want to turn it off.

It's an alpha feature.  Would it be terrible if we made it opt-in and broke that "no config required" default we have today?